src/header.rs

```diff
@@ -868,10 +868,8 @@ impl<W: io::Write> UncompressedHeader for BitWriter<W, BigEndian> {
   fn write_frame_size<T: Pixel>(
     &mut self, fi: &FrameInvariants<T>,
   ) -> io::Result<()> {
-    // width_bits and height_bits will have to be moved to the sequence header OBU
-    // when we add support for it.
-    let width_bits = 32 - (fi.width as u32).leading_zeros();
-    let height_bits = 32 - (fi.height as u32).leading_zeros();
+    let width_bits = fi.sequence.num_bits_width;
+    let height_bits = fi.sequence.num_bits_height;
```

Comment:
You should also write fi.sequence.max_frame_width and fi.sequence.max_frame_height instead of fi.width and fi.height within this function, now that they may differ.